### ISSUE TITLE
Small Elemental Enemies

### ISSUE THREAD

**Issue by Kaimatten** [lancehalberd/spirit-world#37]
I want a trio of small elemental enemies that can be encountered in the spirit world. They are supposed to be ethereal energy beings sort of like very weak versions of the Spirit Beasts. There would be one for each element, fire/ice/lightning. I can have them trail the elemental particles we already use in other places.
I don't think they need any particular action animations currently, although if you are inspired and want to suggest any that is fine. Currently they will probably just float to the player and try to deal contact damage if in range. I guess one animation I have in mind is that they are immune to most types of damage, so having some animation of them fuzzing apart and then coming back together whenever an attack passes through them would be interesting to have.
We could also experiment with giving them different movement style animations if you think of anything appropriate. The ice element for example could be like a trail of growing ice spikes along surfaces and the fire elemental could be some kind of floating flame like the Bomb type enemies from Final Fantasy. Maybe the lightning elemental sticks to solid objects then travels like a bolt of lightning to nearby objects when it moves (probably with some kind of warning so the player has a chance to dodge).
My plan is to have these enemies be able to bond with certain normal enemies to create hybrid enemies with upgraded abilities. I don't know if this impacts the design at all for you, but just mentioning it in case it does.

**Comment by Kaimatten** [comment]
Sounds good. I think first step is for me to concept out elemental monsters, which sounds like they'll all basically just be the element but sentient. Maybe they'll all have faces to help establish they are sentient? (like the bomb that you mention)

What size should they be? 1 tile? Bigger?

**Comment by lancehalberd** [comment]
These should be approximately one tile. Faces wouldn't hurt but also aren't necessary. Really whatever you think is fun/compelling for them for now. I have this in high priority because I plan on adding these soon, but it isn't necessarily a rush, I can just use existing elemental symbols for them as placeholders until we come up with a first draft.

**Comment by Kaimatten** [comment]
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/9da41898-9e51-493a-a920-b0f6ee217901)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/4809b24d-3900-4564-bfd0-9e8e95db95ea)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/25820eda-5fb4-47fd-bc02-efe130aa950d)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/a14682a2-fe24-47de-95c6-ed366dbd21b3)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/7a1248bf-ef1d-4d46-b11d-89098bef9bba)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/f87c5379-e045-4122-9e66-60fe03a2c50f)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/3b38bf6b-97d4-4184-8dc5-8b680802776f)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/8d7e5ca7-692e-4c42-9f46-4ac32912fd0d)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/662add09-156f-4189-9be0-7c6e3d6157dd)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/0d3f2159-788d-479b-82c1-9b61facc7dfc)

Different versions of possible elemental enemies with 16px hitboxes. With and without faces. If any look good, we can move forward with them, otherwise we can discuss changes/different approaches.

**Comment by lancehalberd** [comment]
I think I prefer having the faces for now. Maybe include them in a way that is easy to remove later if I change my mind.
Fire seems good. 

I think I prefer the spike ice one assuming we can make it move by looking like the spikes are growing in a trail. I guess I don't love the second ice one either way. If you sense any trouble with the spikes version, we could try one that looks like a snow flake crystal instead.

I like the ball lightning one, I don't care for the lightning bolt shaped one, although it might be alright animated.

**Comment by Kaimatten** [comment]
For the ice spike, the trail animation may be tricky to put together. Changing directions can mess up mid animations of the trail. You probably have something similar to what I have in mind:
What we can do is have the Ice Spike actually be a bunch of sprites that you have pinned to an invisible moving object.
The invisible moving object spawns the ice spikes, which will have an animation of popping out of the ground and slowly retracting with a slight difference to its X and Y values each time.
Does this make sense or did you have something else that works? Mostly I'm avoiding doing 4 directions of manual ice spike trails animated, as it can cause issues if the ice spike changes directions or wants to move diagonally, etc. Also it just will take way longer.

For the Fire and Lightning, it sounds like the animations we need are just idle, move, and hurt? They don't have a specific attack animation?

**Comment by lancehalberd** [comment]
Whatever seems easiest for the ice spikes. It might be simpler to just animate it in place as spikes rising and falling to sort of suggest that behavior, but then we just move it around like a regular sprite. That would probably be easier for me for now.
The elementals could probably just be a single animation for now if it looks good still and with them moving. Currently they are one hit kills if you have a means to hurt them so even a hurt animation isn't really important. I can always add behaviors to them if you are inspired to animate them doing anything specific.

**Comment by Kaimatten** [comment]
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/7ad9808a-93f9-490a-9ae7-324ff8fe71a3)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/bbceb3b5-726c-45ce-9ea7-c80362ea2b7a)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/ed28ca68-a503-457f-896e-a5d9068a0c81)

**Comment by Kaimatten** [comment]
Ice Spikes are meant to fade away at its last frame, and a new ice spike forms. alternatively, I can make it so it just goes up and down back and forth?

**Comment by lancehalberd** [comment]
Hmm, I think to move the ice spike I'll need to have quite a few on screen, but I think each one having a face will be confusing. If you think you have a good solution to this, we can try that, but otherwise maybe we should just simplify and change it to function more like the others and just have it be a big snowflake with eyes?
If we wanted to keep the current style, the easiest thing might be to just have a single animation of a bunch of spikes rising and falling in place that I just move around like a regular sprite and don't worry about leaving a trail for now. Not exactly sure how the face would be incorporated in that animation though.

**Comment by Kaimatten** [comment]
Snowflake with eyes if better?

![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/ecdcebcc-fa2e-4e4d-af0f-633e39ad52e6)

**Comment by lancehalberd** [comment]
Yeah I think I like that better than the spike with eyes, let's go with that.

**Comment by Kaimatten** [comment]
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/b2eb2d75-b123-4a37-92cb-1540c0ece37a)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/ed936e81-352b-449c-97ab-ce4d4c04080b)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/3a21e0a2-db45-467e-a78d-192735e44bba)
![Image](https://github.com/lancehalberd/spirit-world/assets/141004606/964f5e32-43d6-4ff0-a233-c4841dfa8cc8)

Spinning snowflake work for movement? I made it a little uneven in its hopping spin but I can also make it purely even if you don't like the feel.
Attached the spritesheets for the elementals if the snowflake works.
Fire and Lightning run at 10 FPS.
Ice runs at 5 FPS

**Comment by lancehalberd** [comment]
Thanks, going to see if I can try these out today!

**Comment by lancehalberd** [comment]
Yeah, I think these look great, I'm moving this to done.
